### PR TITLE
EOS-20697: Rewrite pacemaker rules to accommodate new and future requirements

### PR DESCRIPTION
**Pull request by ajaysrivas**
## Problem Statement
<pre>
  <code>
  EOS-20697: HA: during bootstrap motr ioservices have dependency on all confd m0d's
  Added cortx_message_bus service in the rules.
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes - on three node VMs.
  </code>
</pre>
## Solution
<pre>
  <code>
    To implement constraint rules based on number of instances running of a resource, a mechanism is required to count the
    running instances. It was achieved by updating node level attributes using attrd_updater.
    A new HA service srv_counter is introduced for this purpose.
  </code>
</pre>

### CODE REVIEW

**Comment by gowthamchinna** [comment]
test this please

**Comment by gowthamchinna** [comment]
test this please

**Review by indrajitzagade — APPROVED** [review]

File: ha/resource/service_instances_counter.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                out, _, _ = executor.run_cmd(f"attrd_updater -Q -A -n {service}", check_error=False)
+                count += AttribUpdater._get_count_from_output(out, node_name)
+        except Exception as e:
+            Log.error(f"Problem in fetching attr. resource: {resource}. Error: {e}")
+            count = -1
+
+        return count
+
+    @staticmethod
+    def update_attr(scope: str, resource: str, instances_per_node: int, node_name:str = None):
```
[the remark this answers is not in the document]
Comment (ajaysrivas) [reply ha/resource/service_instances_counter.py:84]: These are static functions which executes some commands and have put those inside a class for logical grouping only. The functions in dynamic_fid_service_ra class are not required in other and functions (aggregated value) in this module are not required in dynamic.. But, it is a good idea to have a different module for attrib. Requires design work and We should refactor the code later if adoption of attrib_updater increases.

[the remark this answers is not in the document]
Comment (ArchanaLimaye) [reply ha/resource/service_instances_counter.py:84]: okay.

**Review by ArchanaLimaye — APPROVED** [review]

**Comment by ajaysrivas** [comment]
VM deployment passed - http://eos-jenkins.mero.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/HA/193/console

**Pull request merged**